service/keychain/cri/v1alpha/cri.go

```diff
@@ -81,14 +81,14 @@ type instrumentedAlphaService struct {
 	configMu sync.Mutex
 }
 
-func (in *instrumentedAlphaService) credentials(host string, refspec reference.Spec) (string, string, error) {
+func (in *instrumentedAlphaService) credentials(host string) (string, string, error) {
 	if host == "docker.io" || host == "registry-1.docker.io" {
 		// Creds of "docker.io" is stored keyed by "https://index.docker.io/v1/".
 		host = "index.docker.io"
 	}
 	in.configMu.Lock()
 	defer in.configMu.Unlock()
-	if cfg, ok := in.config[refspec.String()]; ok {
```

Comment:
If we make this change, we also have to update `PullImage` and `DeleteImage` so they update the config using the same key.

I'm not sure this change is correct, though. I don't think there's anything technical that prevents having unique CRI credentials for every image rather than every host.

Reply:
> If we make this change, we also have to update PullImage and DeleteImage so they update the config using the same key.

Ah, good catch

Reply:
Re: https://github.com/awslabs/soci-snapshotter/pull/977#discussion_r1425676753, I'll revert this.